.changeset/twenty-timers-cheat.md

```diff
@@ -0,0 +1,7 @@
+---
+'xstate': patch
+---
+
+State that is persisted and restored from `machine.resolveState(state)` will now have the correct `state.machine` value, so that `state.can(...)` and other methods will work as expected.
+
+See #3096 for more details.
```

Comment:
```suggestion
State that is persisted and restored from `machine.resolveState(state)` will now have the correct `state.machine` value, so that `state.can(...)` and other methods will work as expected. See [#3096](https://github.com/statelyai/xstate/issues/3096) for more details.
```

Reply:
> q: do u want to make a rule that we should always reference the closed ticket in changesets?

Yes, this is a good idea